ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/execution/JsonRpcExecutorVerticle.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.execution;
+
+import static org.hyperledger.besu.ethereum.api.jsonrpc.EventBusAddress.RPC_EXECUTE_ARRAY;
+import static org.hyperledger.besu.ethereum.api.jsonrpc.EventBusAddress.RPC_EXECUTE_OBJECT;
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.EXCEEDS_RPC_MAX_BATCH_SIZE;
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.INTERNAL_ERROR;
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.INVALID_REQUEST;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcConfiguration;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequest;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+
+import io.opentelemetry.api.trace.Tracer;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JsonRpcExecutorVerticle extends AbstractVerticle {
+  private static final Logger LOG = LoggerFactory.getLogger(JsonRpcExecutorVerticle.class);
+
+  private final JsonRpcExecutor jsonRpcExecutor;
+  private final Tracer tracer;
+  final JsonRpcConfiguration jsonRpcConfiguration;
+
+  public JsonRpcExecutorVerticle(
+      final JsonRpcExecutor jsonRpcExecutor,
+      final Tracer tracer,
+      final JsonRpcConfiguration jsonRpcConfiguration) {
+    this.jsonRpcExecutor = jsonRpcExecutor;
+    this.tracer = tracer;
+    this.jsonRpcConfiguration = jsonRpcConfiguration;
+  }
+
+  @Override
+  public void start() {
+    vertx
+        .eventBus()
+        .consumer(RPC_EXECUTE_OBJECT.getAddress(), this::handleJsonRpcExecutorObjectRequest);
+    vertx
+        .eventBus()
+        .consumer(RPC_EXECUTE_ARRAY.getAddress(), this::handleJsonRpcExecutorArrayRequest);
+  }
+
+  private void handleJsonRpcExecutorObjectRequest(
+      final Message<JsonRpcExecutorObjectRequest> requestMessage) {
+    final JsonRpcExecutorObjectRequest request = requestMessage.body();
+    LOG.trace("Received {}", request);
+
+    final JsonRpcResponse jsonRpcResponse =
+        jsonRpcExecutor.execute(
+            request.getOptionalUser(),
+            tracer,
+            request.getSpanContext(),
+            request.getAlive(),
+            request.getJsonObject(),
+            req -> req.mapTo(JsonRpcRequest.class));
+
+    requestMessage.reply(jsonRpcResponse);
+  }
+
+  private void handleJsonRpcExecutorArrayRequest(
+      final Message<JsonRpcExecutorArrayRequest> requestMessage) {
+    final JsonRpcExecutorArrayRequest request = requestMessage.body();
+    final JsonArray jsonArray = request.getJsonArray();
+
+    if (jsonRpcConfiguration.getMaxBatchSize() > 0
```

Comment:
this would mean if I set maxBatchSize = 0 it would be effectively unlimited, right?

Reply:
I think this is the same as the current behaviour - 
-1 is unlimited
>=1 sets a limit
0 doesn't really make sense so maybe we should just doc that 0 is unlimited, same as -1